### PR TITLE
Filter suppliers on a framework

### PR DESCRIPTION
**Pull request by robyoung**
Add the route `/frameworks/<framework_slug>/suppliers` for listing suppliers associated with a framework. This expands on what is provided through `/frameworks/<framework_slug>/interest` with full supplier framework objects and allows the list to be filtered by whether they have returned their agreement.

This is required for the admin interface for CCS which will allow them to list suppliers that have returned their framework agreements with the most recent first.

### CODE REVIEW

File: app/main/views/frameworks.py [hunk 1]
```diff
@@ -136,6 +137,33 @@ def label_columns(labels, query):
...
+            supplier_frameworks = supplier_frameworks.filter(
+                SupplierFramework.agreement_returned == false()
+            )
+
+    return jsonify(supplierFrameworks=[
+        supplier_framework.serialize() for supplier_framework in supplier_frameworks
+    ])
+
+
 @main.route('/frameworks/<string:framework_slug>/interest', methods=['GET'])
```
Comment (TheDoubleK) [inline app/main/views/frameworks.py]: Should we deprecate this `interest` endpoint in favour of the new one? (I'm not saying make all the changes now! But to tidy up at some later date.)

File: app/main/views/frameworks.py [hunk 2]
```diff
@@ -136,6 +137,33 @@ def label_columns(labels, query):
...
+    agreement_returned = request.args.get('agreement_returned')
+
+    supplier_frameworks = SupplierFramework.query.filter(
+        SupplierFramework.framework_id == framework.id
+    )
+
+    if agreement_returned is not None:
+        if convert_to_boolean(agreement_returned):
+            supplier_frameworks = supplier_frameworks.filter(
+                SupplierFramework.agreement_returned == true()
```
Comment (TheDoubleK) [inline app/main/views/frameworks.py]: By having `== true()` here does it mean that both `?agreement_returned=true` and  `?agreement_returned=false` will return the list of suppliers who have returned their agreement, which seems odd.

Should it be `== convert_to_boolean(agreement_returned)` instead to get either suppliers who have or suppliers who haven't returned their agreements?  Or have I missed something?

File: tests/app/views/test_suppliers.py [hunk 3]
```diff
@@ -1,5 +1,8 @@
+from datetime import datetime
 from flask import json
-from nose.tools import assert_equal, assert_in, assert_is_not_none, assert_true
+import pytest
+from freezegun import freeze_time
```
Comment (TheDoubleK) [inline tests/app/views/test_suppliers.py]: I like this `freeze_time` - nice find.

**Comment by TheDoubleK** [comment]
Reviewed - looks good, other than being able to filter on `agreement_returned=false`

File: app/main/views/frameworks.py (hunk 1 above)
Comment (robyoung) [inline app/main/views/frameworks.py]: Yea, I was going to do that when I rename the response key.

File: app/main/views/frameworks.py (hunk 2 above)
Comment (robyoung) [inline app/main/views/frameworks.py]: Talked in real world.

**Comment by TheDoubleK** [comment]
:+1: once rebased

**Pull request merged**